platform.c

```diff
@@ -0,0 +1,18 @@
+#include <errno.h>
+#include <unistd.h>
+
+#include "platform.h"
+#include "logging.h"
+
+int init_platform_page_size(void)
+{
+    if (!platform_page_size) {
+        long result = sysconf(_SC_PAGESIZE);
+        if (result < 0) {
+            VHD_LOG_ERROR("failed to get page size by sysconf: %ld errno %d", result, errno);
+            return -1;
```

Comment:
Why not make this just propagate errno directly?

Reply:
fixed